Recipely/Recipely/LoginScreen/Presenter/LoginPresenter.swift

```diff
@@ -0,0 +1,98 @@
+// LoginPresenter.swift
+// Copyright © RoadMap. All rights reserved.
+
+import UIKit
+
+protocol LoginViewControllerProtocol: AnyObject {
+    func setValidationStatusEmail(_ colorText: UIColor, _ colorBorder: UIColor, _ isValidation: Bool)
+    func setValidationStatusPassword(_ colorText: UIColor, _ colorBorder: UIColor, _ isValidation: Bool)
+    func setloginButtonPressed(_ nameOfButtonImage: String, _ titleButton: String)
+    func showErrorSplashOn()
+    func showErrorSplashOff()
+    func showPassword(_ eyeImageView: UIImage, _ isSecurity: Bool)
+}
+
+protocol LoginPresenterProtocol: AnyObject {
+    func setColorEmail(_ email: String)
+    func setColorPassword(_ password: String, _ email: String)
+    func setSecurity(_ isSecurity: Bool)
+}
+
+/// Презентер экрана авторизации
+final class LoginPresenter: LoginPresenterProtocol {
+    // MARK: - Constants
+
+    private enum Constants {
+        static let domenEmail = "@mail.ru"
+        static let errorColor = "errorColor"
+        static let defaultTextColor = "defaultTextColor"
+        static let defaultBorderTextFieldColor = "defaultBorderTextFieldColor"
+        static let eyeImageName = "eye.fill"
+        static let eyeSlashImageName = "eye.slash.fill"
+        static let spinerIamgename = "spiner"
+        static let titleButtonText = "Login"
+        static let dispatchQueueTime = 3.0
+        static let passwordCount = 5
+    }
+
+    // MARK: - Public Properties
+
+    weak var loginCoordinator: LoginCoordinator?
+    weak var view: LoginViewControllerProtocol?
+
+    // MARK: - Initializers
+
+    required init(view: LoginViewControllerProtocol) {
```

Comment:
```suggestion
    required init(view: LoginViewControllerProtocol) {
```
если ты иньектишь через конструктор тогда  loginCoordinator и  view свойства должны быть private и все передавай через init